### PR TITLE
build(deps): Upgrade setup-go action to use go.mod as a version file

### PR DESCRIPTION
**Pull request by abtris**
Signed-off-by: Ladislav Prskavec <ladislav@prskavec.net>

### CODE REVIEW

File: .github/workflows/integration_tests.yml
Comment (christian-kreuzberger-dtx) [inline .github/workflows/integration_tests.yml:32]: Here we still need to move the checkout before setup-go

Comment (abtris) [reply .github/workflows/integration_tests.yml:27]: Fixed

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM

**Comment by Raffy23** [comment]
Hi @abtris, it seems there are some merge conflicts that must be resolved before merging. After that, we can approve the PR again so that you can merge it into main.

**Comment by abtris** [comment]
@Raffy23 rebased with current main

**Comment by abtris** [comment]
@Raffy23 I don't have the right to merge it.

**Comment by Raffy23** [comment]
Sorry, I wasn't aware that you can't merge the request, I will merge it for you.

**Pull request merged**
